portal/src/components/leaderboard/NewLeaderboard.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Flex, Heading, Spacer, Button } from "@chakra-ui/react";
+import LeaderboardFields from "./LeaderboardFields";
+
+const NewLeaderboard = ({ handleCreate, onClose }) => {
+  const [title, setTitle] = React.useState("");
+  const [description, setDescription] = React.useState("");
+
+  return (
+    <>
+      <Heading fontSize={["lg", "2xl"]}>Create Leaderboard</Heading>
+      <LeaderboardFields
+        title={title}
+        setTitle={setTitle}
+        description={description}
+        setDescription={setDescription}
+      />
+      <Spacer />
+      <Flex alignSelf="flex-end">
+        <Button w="135px" bgColor="#4D4D4D" color="inherit" mx="10px" onClick={onClose}>
+          Cancel
+        </Button>
+        <Button
+          w="200px"
+          bgColor="#F56646"
+          color="inherit"
+          mx="10px"
+          onClick={async () => {
+            await handleCreate(title, description);
+            onClose();
+          }}
+        >
+          Create
+        </Button>
```

Comment:
No indication that something is happening. For POST requests I use useMutation to have request status and to handle success and error. 
Like this:
```
              <Button
                  onClick={() =>
                    setPoolController.mutate({ controller: newPoolController, poolId: selectedPool })
                  }
                  isDisabled={setPoolController.isLoading}
              >
                {setPoolController.isLoading ? <Spinner /> : "Save"}
              </Button>
```

useMutation is very handy, It even has call's arguments inside so if you have same buttons you can tell which one triggered the action:
            <Box
              className={styles.button}
              color={"#FFF"}
              cursor={claimDrop.isLoading ? "not-allowed" : "pointer"}
              onClick={() => {
                if (!claimDrop.isLoading) {
                  claimDrop.mutate({ request });
                }
              }}
            >
              {claimDrop.isLoading && claimDrop.variables?.request === request ? (
                <Spinner h={{ base: "16px", sm: "18px" }} w={{ base: "16px", sm: "18px" }} />
              ) : (
                "Claim"
              )}
            </Box>